### ISSUE TITLE
💅 Astro comment suppressions report unknownRule for noAstroSetHtmlDirective

### ISSUE THREAD

**Issue by finnan444** [biomejs/biome#11653]
Suppressing `noAstroSetHtmlDirective` from an Astro comment does not work: the suppression itself is reported as `suppressions/unknownRule`, and the rule still fires on the next line. The same suppression written as an HTML comment works.

This matters because the HTML comment survives into the built output, so the suppression ends up published on the page, while the Astro comment (which Astro strips at build time) is the one that gets rejected.

### Environment information

<details>

```bash
CLI:
  Version:                      2.5.12
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:                npm
  JS_RUNTIME_VERSION:                v26.7.0
  JS_RUNTIME_NAME:                   node

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    true

Linter:
  Recommended:                  false
  Enabled rules:
    nursery/noAstroSetHtmlDirective
```

</details>

### Rule name

`nursery/noAstroSetHtmlDirective`

### Reproduction

The playground does not accept `.astro` files, so here is a minimal local repro instead.

`biome.json`:

```json
{
  "html": { "experimentalFullSupportEnabled": true },
  "linter": {
    "rules": {
      "recommended": false,
      "nursery": { "noAstroSetHtmlDirective": "error" }
    }
  }
}
```

`astro.astro` - suppression via an Astro comment:

```astro
{/* biome-ignore lint/nursery/noAstroSetHtmlDirective: trusted constants */}
<script is:inline type="application/ld+json" set:html={JSON.stringify({ name: "Example" })} />
```

`html.astro` - same thing via an HTML comment:

```astro
<!-- biome-ignore lint/nursery/noAstroSetHtmlDirective: trusted constants -->
<script is:inline type="application/ld+json" set:html={JSON.stringify({ name: "Example" })} />
```

`biome lint astro.astro` exits 1 with two diagnostics:

> astro.astro:1:2 suppressions/unknownRule
> ! Unknown lint rule nursery/noAstroSetHtmlDirective in suppression comment

> astro.astro:2:46 lint/nursery/noAstroSetHtmlDirective
> × The set:html directive inserts unescaped HTML.

`biome lint html.astro` exits 0.

### Expected result

`{/* biome-ignore ... */}` should suppress the diagnostic on the following element, the same way the HTML comment does, and should not report `suppressions/unknownRule`.

### Notes

Plain Astro comments are handled fine elsewhere; the problem is specific to suppression comments. My guess is that the Astro comment is parsed as a JS/expression-flavoured comment and looked up against a rule registry that does not include the HTML rules, but I have not verified that in the source.

#8294 ("Astro comments aren't correctly recognized") looks related and is closed, but I could not confirm whether the linked PRs cover this suppression path, so I am filing separately rather than claiming it is the same bug.

Practical impact: the only working suppression leaks a comment into shipped HTML, so the alternative is a per-file `overrides` entry that turns the rule off for the whole component. Neither is great for a single `<script type="application/ld+json">` with trusted JSON-LD, which is the documented use of `set:html` (https://docs.astro.build/en/reference/directives-reference/#sethtml).

**Comment by biome-cookie** [comment]
Hello @finnan444, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by ematipico** [comment]
> The playground does not accept .astro files, so here is a minimal local repro instead.

[It does ](https://biomejs.dev/playground/?lintRules=noMisusedPromises&tab=formatter&pane=Diagnostics&language=astro#code=LQAtAC0ACgBsAGUAdAAgAGYAbwBvADsACgAtAC0ALQAKAAoAPABkAGkAdgA%2BADwALwBkAGkAdgA%2BAA%3D%3D)

**Comment by finnan444** [comment]
My mistake, sorry
Astro comment (broken): https://biomejs.dev/playground/?lintRules=noAstroSetHtmlDirective&language=astro&pane=Diagnostics#code=ewAvACoAIABiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAgAGwAaQBuAHQALwBuAHUAcgBzAGUAcgB5AC8AbgBvAEEAcwB0AHIAbwBTAGUAdABIAHQAbQBsAEQAaQByAGUAYwB0AGkAdgBlADoAIAB0AHIAdQBzAHQAZQBkACAAYwBvAG4AcwB0AGEAbgB0AHMAIAAqAC8AfQAKADwAcwBjAHIAaQBwAHQAIABpAHMAOgBpAG4AbABpAG4AZQAgAHQAeQBwAGUAPQAiAGEAcABwAGwAaQBjAGEAdABpAG8AbgAvAGwAZAArAGoAcwBvAG4AIgAgAHMAZQB0ADoAaAB0AG0AbAA9AHsASgBTAE8ATgAuAHMAdAByAGkAbgBnAGkAZgB5ACgAewAgAG4AYQBtAGUAOgAgACIARQB4AGEAbQBwAGwAZQAiACAAfQApAH0AIAAvAD4ACgA%3D

HTML comment (works, control): https://biomejs.dev/playground/?lintRules=noAstroSetHtmlDirective&language=astro&pane=Diagnostics#code=PAAhAC0ALQAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAG4AdQByAHMAZQByAHkALwBuAG8AQQBzAHQAcgBvAFMAZQB0AEgAdABtAGwARABpAHIAZQBjAHQAaQB2AGUAOgAgAHQAcgB1AHMAdABlAGQAIABjAG8AbgBzAHQAYQBuAHQAcwAgAC0ALQA%2BAAoAPABzAGMAcgBpAHAAdAAgAGkAcwA6AGkAbgBsAGkAbgBlACAAdAB5AHAAZQA9ACIAYQBwAHAAbABpAGMAYQB0AGkAbwBuAC8AbABkACsAagBzAG8AbgAiACAAcwBlAHQAOgBoAHQAbQBsAD0AewBKAFMATwBOAC4AcwB0AHIAaQBuAGcAaQBmAHkAKAB7ACAAbgBhAG0AZQA6ACAAIgBFAHgAYQBtAHAAbABlACIAIAB9ACkAfQAgAC8APgAKAA%3D%3D

The only difference between the two is the comment syntax. On 2.5.12 the CLI gives `suppressions/unknownRule` plus the unsuppressed `noAstroSetHtmlDirective` error for the Astro comment, and exit 0 for the HTML comment.
